vochain/process_id.go

```diff
@@ -0,0 +1,120 @@
+package vochain
+
+import (
+	"encoding/binary"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"go.vocdoni.io/dvote/crypto/ethereum"
+	"go.vocdoni.io/proto/build/go/models"
+	"google.golang.org/protobuf/encoding/protowire"
+)
+
+// ProcessID is a 32 bytes identifier that holds the following information about the voting process:
+//
+// - chainID: a 6 bytes trunked hash of the blockchain identifier
+//
+// - organization: hex address of the wallet creating the process
+//
+// - census proof type: kind of census used for the process (from protobuf models)
+//
+// - nonce: an incremental uint32 number
+//
+// processID: 395b41cc9abcd8da6bf26964af9d7eed9e03e53415d37aa96045000600000001 represents
+// chID:vocdoni-azeno addr:0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045 pt:arbo nc:1
+type ProcessID struct {
+	chainID          string
+	organizationAddr *common.Address
+	proofType        uint16
+	nonce            uint32
+}
+
+// Marshal encodes to bytes
+func (p *ProcessID) Marshal() []byte {
+	//processID = chainID[6bytes] + organizationAddress[20bytes] + proofType[2bytes] + nonce[4bytes]
+	chainID := ethereum.HashRaw([]byte(p.chainID))
+
+	nonce := make([]byte, 4)
+	binary.BigEndian.PutUint32(nonce, uint32(p.nonce))
+
+	proofType := make([]byte, 2)
+	binary.BigEndian.PutUint16(proofType, p.proofType)
+
+	id := make([]byte, 32)
+	if p.organizationAddr == nil {
+		panic("organization cannot be nil")
+	}
+	id = append(chainID[:6], p.organizationAddr.Bytes()[:20]...)
+	id = append(id, proofType[:2]...)
+	return append(id, nonce[:4]...)
+}
+
+// Unmarshal decodes a 32 byte payload into the process ID fields
+func (p *ProcessID) Unmarshal(pid []byte) error {
+	if len(pid) != 32 {
+		return fmt.Errorf("process ID lenght not correct")
+	}
+	p.chainID = fmt.Sprintf("%x", pid[:4])
+	baddr := make([]byte, 20)
+	copy(baddr[:], pid[6:26])
+	addr := common.BytesToAddress(baddr)
+	p.organizationAddr = &addr
+	p.proofType = binary.BigEndian.Uint16(pid[26:28])
+	p.nonce = binary.BigEndian.Uint32(pid[28:32])
+	return nil
+}
+
+// String returns a human readable representation of process ID
+func (p *ProcessID) String() string {
+	return fmt.Sprintf("[chID:%s/addr:%s/pt:%s/nc:%d]",
+		p.chainID, p.organizationAddr.Hex(), p.Proof(), p.nonce)
+
+}
+
+// SetChainID sets the blockchain identifier
+func (p *ProcessID) SetChainID(chID string) {
+	p.chainID = chID
+}
+
+// ChainID returns blockchain identifier
+func (p *ProcessID) ChainID() string {
+	return p.chainID
+}
+
+// SetAddr sets the organizer address (20 bytes)
+func (p *ProcessID) SetAddr(addr common.Address) {
+	p.organizationAddr = &addr
+}
+
+// Addr returns the organizer address
+func (p *ProcessID) Addr() *common.Address {
```

Comment:
per the above, I don't think this should return a pointer (SetAddr is already good)